module/zfs/dsl_dataset.c

```diff
@@ -2386,8 +2386,8 @@ get_receive_resume_token_impl(dsl_dataset_t *ds)
 	}
 	if (dsl_dataset_feature_is_active(ds,
 	    SPA_FEATURE_REDACTED_DATASETS)) {
-		uint64_t num_redact_snaps;
-		uint64_t *redact_snaps;
+		uint64_t num_redact_snaps = 0;
+		uint64_t *redact_snaps = NULL;
 		VERIFY(dsl_dataset_get_uint64_array_feature(ds,
 		    SPA_FEATURE_REDACTED_DATASETS, &num_redact_snaps,
```

Comment:
The `VERIFY()` usage here isn't quite consistent with other callers and I wonder if it's responsible for suppressing the error.  Is the error logged properly if you make the following change?  There is some discussion of this in the [libtool documentation](https://www.gnu.org/software/libtool/manual/libtool.html#Compile-mode). 

```suggestion
 		VERIFY3B(dsl_dataset_get_uint64_array_feature(ds,
		    SPA_FEATURE_REDACTED_DATASETS, &num_redact_snaps,
     		    &redact_snaps, ==, B_TRUE));
```

Either way the fix looks good.  It looks like we should make the same fix to the `uint64_t num_redact_snaps, int_size;` variables in the next conditional a block a few lines down.  I noticed these are in a `VERIFY0` which maybe also somehow ate the warnings I would have expected.

Reply:
I found that document before, but I don't immediately see where you're referring to in particular.

I'm all in favor of turning unmoderated VERIFYs into things that actually print useful errors, though, even though as it turns out, changing that into a VERIFY3B does _not_ cause it to actually print. (I'd have been surprised if it did, I think, since I don't think one of the objects compiles VERIFYs out and the other doesn't?)

Reply:
Sure, I actually did that, but then forgot to push after finding the VERIFY3B change didn't help (which I threw in with it, because IMO plain VERIFYs should go away everywhere possible).